editors/code/src/commands/cargo_watch.ts

```diff
@@ -0,0 +1,137 @@
+import * as child_process from 'child_process';
+import * as path from 'path';
+import * as timers from 'timers';
+import * as vscode from 'vscode';
+import { StatusDisplay } from './watch_status';
+
+export class CargoWatchProvider {
+    private diagnosticCollection?: vscode.DiagnosticCollection;
+    private cargoProcess?: child_process.ChildProcess;
+    private outBuffer: string = '';
+    private statusDisplay?: StatusDisplay;
+
+    public activate(subscriptions: vscode.Disposable[]) {
+        subscriptions.push(this);
+        this.diagnosticCollection = vscode.languages.createDiagnosticCollection(
+            'rustc'
+        );
+
+        this.statusDisplay = new StatusDisplay(subscriptions);
+
+        // Start the cargo watch with json message
+        this.cargoProcess = child_process.spawn(
+            'cargo',
+            ['watch', '-x', '"check --message-format json"'],
```

Comment:
while we are at it, let's maybe add a `cargo watch command` config, such that user's can enable/disable features, etc?

Reply:
Sure, and i will propose the following config :

`rust-analyzer.cargo-watch.check-arguments` 

which corresponding to :

```typescript
  this.cargoProcess = child_process.spawn(
            'cargo',
            ['watch', '-x', '"check --message-format json ${this.check_arguments}"'  ],
```
And i want to add an commands for restarting the whole cargo-watch process. Just like `Restart RLS` in `rls-vscode`. What do you think about it ?

Reply:
Config seems good to me.

As for commands, they seems useful, but not critical: it's ok to start without them.